### ISSUE TITLE
ConstantName should allow CamelCase constants for class aliases

### ISSUE THREAD

**Issue by marcandre** [bbatsov/rubocop#5053]
The follow code is flagged by RuboCop:

```
class Foo
end
Bar = Foo
```

I believe it shouldn't complain about `Bar` being camelcase here. Assuming `Foo` is correctly written, then `Bar` *should* actually be CamelCase, and `BAR = Foo` should actually be an error.

I understand there is rarely a case for such code (but we have nevertheless 14 such cases 😈 )

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.3.5 x86_64-darwin15)
```

**Comment by garettarrowood** [comment]
This sounds like a fun fix. Happy to take a stab at it over the next few days.

**Comment by garettarrowood** [comment]
As that comment suggestions, it looks like the described behavior actually is not flagged by rubocop. I tried to reproduce without success. Adding this spec also passes:

```
  # spec/rubocop/cop/naming/constant_name_spec.rb
  it 'does not throw offense if rhs is camelCase' do
    expect_no_offenses(<<-RUBY.strip_indent)
      class FooBaz
      end
      BarBaz = FooBaz
    RUBY
  end
```

@marcandre - Can you provide any additional info about error?

**Comment by marcandre** [comment]
Indeed, `Foo = Bar` is ok, but `Foo = Bar = Qux` is flagged.

**Comment by marcandre** [comment]
Thanks for the fix. On the other hand, this allows `Foo = BAR` and `Bar = FOO`, right?

**Comment by garettarrowood** [comment]
From a quick test, it looks like those examples work just fine. No offenses on my end.

**Comment by marcandre** [comment]
Yeah. Maybe it should be an offense though, at least `Foo = BAR`. Not sure if `FOO = Bar` is ok.
